### DISCUSSION TITLE
How to get Templ Proxy Server working from inside a docker container

### DISCUSSION THREAD

**Discussion by grantralls** [a-h/templ#683] [Q&A]
I'm trying to create a development docker compose file so I can stand up all my services in one command reliably. A golang server, an nginx server, and a database. I can get the templ hot reload proxy working without docker, but I can't seem to get it work inside the container. My command is `templ generate --watch --open-browser=false --proxy=http://localhost:8080 --proxybind=localhost -v --cmd="go run ."`. I attempt to connect to the server from my browser with `http://localhost:7331`, I immediately get "site cannot be reached". With no output in the container console. I attempt to hit `http://localhost:8080` and I get results as I expect but of course that's essentially circumventing the proxy server. I've made sure I've exposed and mapped the proper ports but still no luck.

Any ideas on how I should proceed?

**Comment by llamux** [comment]
Have you tried using `--proxybind=0.0.0.0`?

**Reply by grantralls** [reply]
So that worked in terms of letting me connect to :7331, but now reloads aren't working right. The console shows "Sending reload event" when I save a change but my browser doesn't refresh. If I connect directly to :7331 reloads work as expected. But I'm currently setup for requests to go to nginx, it makes an attempt to serve a static file, if it can't, the request gets sent to the backend server which is the templ proxy.

So if I use the templ proxy directly it works (but I don't get my static files of course), but if I use the nginx server I get my static files but the templ proxy doesn't automatically refresh.

**Reply by llamux** [reply]
Would you be able to link a minimal example repo with your nginx.conf, Dockerfile, docker-compose.yaml, etc that reproduces the issue?

**Reply by grantralls** [reply]
https://github.com/bythecover/backend

## Reproduction Steps
1. Clone the repo and have the most up to date version of docker & docker compose installed
2. Change directory to the backend/ folder
3. run `docker compose up`
4. open `localhost:7331/test` in a browser
5. open `localhost:80/test` in a different browser window
6. move the browser windows side by side
7. make a change to page-template.templ then save
8. The console that is running the app should print `Sending reload event`. On my machine, The window running from :7331 gets the hot reload. The window running :80 does not.

I've also just noticed when using :7331 stylesheets do not get applied. When using 7331 I changed the url for the stylesheet to be `localhost:80/output.css`

**Reply by a-h** [reply]
The underlying issue is likely to be https://github.com/fsnotify/fsnotify/issues/292 - fsnotify doesn't work on Windows, and I think there's issues with Docker too.

Hugo has a fallback mode that uses polling as per https://github.com/gohugoio/hugo/tree/master/watcher/filenotify

Air seems to use the Hugo implementation as per https://github.com/cosmtrek/air/pull/409/files

So options are:

- Avoid Docker (if that's the problem)
- Switch to Air
- Add a polling fallback to the templ hot reload
